graspologic/nominate/VNviaSGM.py

```diff
@@ -0,0 +1,391 @@
+import itertools
+import warnings
+
+
+class VNviaSGM(BaseEstimator):
+    """
+    This class implements Vertex Nomination via Seeded Graph Matching
+    (VNviaSGM) with the algorithm described in [1].
+
+    VNviaSGM is a nomination algorithm, so instead of completely matching two
+    graphs `A` and `B`, it proposes a nomination list of potential matches in
+    graph `B` to a vertex of interest (VOI) in graph `A` with associated
+    probabilities. VNviaSGM generates an initial induced subgraph about the
+    VOI to determine which seeds are close enough to be used.
+    All the seeds that are close enough are then used to generate subgraphs in
+    both A and B. These subgraphs are matched using the seeded graph matching
+    algorithm (SGM), and a nomination list is returned. See
+    :class:`~graspologic.match.GraphMatch` for SGM docs.
+
+    Parameters
+    ----------
+    order_voi_subgraph: int, positive (default = 1)
+        Order used to create induced subgraph on `A` about VOI where the max
+        distance between VOI and other nodes is ``order_voi_subgraph``.
+        This induced subgraph will be used to determine what seeds are used
+        when the SGM algorithm is called. If no seeds are in this subgraph
+        about VOI, then a UserWarning is thrown, and ``nomination_list_``
+        is None.
+
+    order_seeds_subgraph: int, positive (default = 1)
+        Order used to  create induced subgraphs on `A`
+        and `B`. These subgraphs are centered about the seeds that were determined
+        by the subgraph generated by ``order_voi_subgraph``. These two subgraphs will
+        be passed into the SGM algorithm.
+
+    n_init: int, positive (default = 100)
+        Number of restarts for soft seeded graph matching algorithm. Increasing the
+        number of restarts will make the probabilities returned more precise.
+
+    max_nominations: int (default = None)
+        Max number of nominations to include in the nomination list. If None is
+        passed, then all nominations computed will be returned.
+
+    graph_match_kws : dict (default = {})
+        Gives users the option to pass custom arguments to the graph matching
+        algorithm. Format should be {'arg_name': arg_value, ...}. See
+        :class:`~graspologic.match.GraphMatch`
+
+    Attributes
+    ----------
+    n_seeds_: int
+        Number of seeds passed in `seedsA` that occured in the induced subgraph about VOI
+
+    nomination_list_: 2d-array
+        An array containing vertex nominations in the form nomination list = [[j, p_val],...]
+        where p_val is the probability that the VOI matches to node j in graph B (sorted by
+        descending probability)
+
+
+    References
+    ----------
+    .. [1] Patsolic, HG, Park, Y, Lyzinski, V, Priebe, CE. Vertex nomination via seeded graph matching. Stat Anal Data
+        Min: The ASA Data Sci Journal. 2020; 13: 229– 244. https://doi.org/10.1002/sam.11454
+
+
```

Comment:
```suggestion
Notes
-----
```

Reply:
@asaadeldin11 should I add notes here or just leave this blank?